src/interpreter/terminator.rs

```diff
@@ -240,6 +240,7 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             // The discriminant_value intrinsic returns 0 for non-sum types.
             Array { .. } | FatPointer { .. } | Scalar { .. } | Univariant { .. } |
             Vector { .. } => 0,
+            UntaggedUnion { .. } => unimplemented!(),
```

Comment:
I think it should be part of the previous arm and return 0, since unions are non-sum types. @eddyb, can you confirm?